src/NemoStuff.jl

```diff
@@ -86,8 +117,129 @@ function is_symmetric(M::MatElem)
     return true
 end
 
+################################################################################
+#
+#  Zero matrix constructors
+#
+################################################################################
+
 zero_matrix(::Type{Int}, r, c) = zeros(Int, r, c)
 
+function zero_matrix(::Type{MatElem}, R::Ring, n::Int)
+    return zero_matrix(R, n)
+end
+
+function zero_matrix(::Type{MatElem}, R::Ring, n::Int, m::Int)
+    return zero_matrix(R, n, m)
+end
+
+function matrix(A::Matrix{T}) where {T<:RingElem}
+    r, c = size(A)
+    (r < 0 || c < 0) && error("Array must be non-empty")
+    m = matrix(parent(A[1, 1]), A)
+    return m
+end
+
+function matrix(A::Vector{T}) where {T<:RingElem}
+    return matrix(reshape(A, length(A), 1))
+end
+
+export scalar_matrix
+
+function scalar_matrix(R::Ring, n::Int, a::RingElement)
+    b = R(a)
+    z = zero_matrix(R, n, n)
+    for i in 1:n
+        z[i, i] = b
+    end
+    return z
+end
+
+function identity_matrix(::Type{MatElem}, R::Ring, n::Int)
+    return identity_matrix(R, n)
+end
+
+function is_zero_row(M::Matrix{T}, i::Int) where {T<:Integer}
+    for j = 1:Base.size(M, 2)
```

Comment:
I'd prefer this, but then we need `ncols(M::Matrix) = Base.size(M, 2)` (which of course we already do have in `Nemo.jl/src/HeckeMoreStuff.jl:1711`...)
```suggestion
    for j = 1:ncols(M)
```

Reply:
`ncols` is part of what is being moved, so no problem in applying this change.